### PR TITLE
Wrote bubble sort algorithm

### PR DESCRIPTION
**Pull request by MiguelDP4**
Wrote bubble sort algorithm for both integers and strings.

### CODE REVIEW

File: bubble.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+def bubble_sort(array)
```
Comment (Forison) [inline bubble.rb:1]: * [ ] bubble_sort method is implemented and working ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
 Currently, ```bubble_sort``` is returning nil, hence any example i try also returns nil
```
int_array = [7, 5, 8, 3, 4, 6, 2, 4, 15, 2, 23]
p bubble_sort(int_array) = nil
```
Let us fix this

File: bubble.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      if array[j].to_i > array[j + 1].to_i
+        temp = array[j]
+        array[j] = array[j + 1]
+        array[j + 1] = temp
+        not_sorted = true
+      end
+      j += 1
+    end
+    i += 1
+  end
```
Comment (Forison) [inline bubble.rb:18]: Return your sorted array here, currently, you are returning nil

File: bubble.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+end
+
+def bubble_sort_by(array)
+  i = 0
```
Comment (Forison) [inline bubble.rb:23]: * [ ] bubble_sort_by method is implemented and working  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
currently for this 
```string_array = %w[hi hello hey another_string haloo testing]
p bubble_sort_by(string_array) { |string1, string2| string1.length <=> string2.length }example ```bubble_sort_by``` 
```
returns
``` 
["hi", "hey", "hello", "haloo", "testing", "another_string"]nil
``` 
look closely at the nil at the back of the sorted array. It ought not be there, kindly fix it.

File: bubble.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        temp = array[j]
+        array[j] = array[j + 1]
+        array[j + 1] = temp
+        not_sorted = true
+      end
+      j += 1
+    end
+    i += 1
+  end
+  print array
```
Comment (Forison) [inline bubble.rb]: let us return an array instead of printing array, note method ought to return not print

**Review by Forison — CHANGES_REQUESTED** [review]
## Status: Require Changes🛑
# 🙋‍♂️ Team,
good job done so far, here are some issues we have to address to get this project approved.
## 👇Check below comments👇
## That is kindly make these changes and submit the project for review
#### Happy coding
#### Addo Forison

**Review by fatymahmed — APPROVED** [review]
### Hi @MiguelDP4 @bren2102 
### Great job making all the suggested changes🎉🎉
### It's a 👍🏾 from me!
## ✔️ Approved!

Good luck with your next projects.
 _Don't forget to submit your project completion form_

File: bubble.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+print bubble_sort(int_array)
+puts ' '
+
+string_array = %w[hi hello hey another_string haloo testing]
+print 'Unordered string array: '
+print string_array
+puts ' '
+print 'Ordered string array: '
+puts ' '
+print bubble_sort_by(string_array) { |string1, string2| string1.length - string2.length }
```
Comment (fatymahmed) [inline bubble.rb:58]: [Optional]
- Please remove this code, you could place it in a different file for the purpose of testing the methods.

**Pull request merged**